packages/shared/src/components/GrowthBookProvider.tsx

```diff
@@ -0,0 +1,127 @@
+import React, {
+  ReactElement,
+  ReactNode,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { useMutation } from 'react-query';
+import {
+  Context,
+  GrowthBook,
+  GrowthBookProvider as Provider,
+} from '../lib/growthbook';
+import { isProduction } from '../lib/constants';
+import { BootApp, BootCacheData } from '../lib/boot';
+import { decrypt } from './crypto';
+import { apiUrl } from '../lib/config';
+
+export type GrowthBookProviderProps = {
+  app: BootApp;
+  user: BootCacheData['user'];
+  deviceId: string;
+  version?: string;
+  experimentation?: {
+    f: string;
+    e: string[];
+  };
+  fetchMethod?: typeof fetch;
```

Comment:
Ugh, really time we context this 😅

Reply:
We have a hook that handles this. Should be named `useRequestProtocol`. It can return the `fetchMethod` which is set automatically by the companion when needed.

Reply:
Ah yes, totally forgot about that, that should do neatly for this purpose.

Reply:
I'll check that out!